### PR TITLE
#793 Documentation updates

### LINKED ISSUE #793: Documentation for recent changes
I started documenting some of it, but need to go through and make sure developer docs are up to date for all recent PRs.

### PR DESCRIPTION
**Pull request by CarlosNZ**
Closes #793 

Gets the internal reference docs up to date with the latest changes, in particular "preview" and "extend" endpoint, plus a few small extras and things that have been missed previously

### CODE REVIEW

**Comment by andreievg** [comment]
modifyMultipleRecords

**Comment by CarlosNZ** [comment]
> did we end up adding modifyMultipleRecords actions to action list docs ?

Oh yeah, thanks for the reminder. Was just going off memory here. Will add that.

**Comment by CarlosNZ** [comment]
> > did we end up adding modifyMultipleRecords actions to action list docs ?
> 
> Oh yeah, thanks for the reminder. Was just going off memory here. Will add that.

1f062887

File: documentation/List-of-Action-plugins.md [hunk 1]
```diff
@@ -219,6 +220,28 @@ It is recommended to use the `data` parameter object when possible. The standalo
...
+
+### Modify Multiple Records
+
+- Allows creating or updating *multiple* database records. It basically just calls the above `modifyRecord` action multiple times.
+
+- _Action Code:_ **`modifyMultipleRecords`**
+
+| Input parameters<br />(\*required) <br/>       | Output properties      |
+| ---------------------------------------------- | ---------------------- |
+| `tableName`                                    | `records: [<results>]` |
```
Comment (andreievg) [inline documentation/List-of-Action-plugins.md:233]: keyMap also I think

Comment (CarlosNZ) [reply documentation/List-of-Action-plugins.md:233]: Good point, thanks: 13afa577

**Comment by nmadruga** [comment]
Probably good to have that also merged to the to-be version 0.4.3, @CarlosNZ? I'll have a quick look now

File: documentation/API.md [hunk 2]
```diff
@@ -352,6 +383,88 @@ For displaying custom data (e.g. Users, Products, Orgs). User's JWT determines w
...
+
+It works by finding an event in the `trigger_schedule` table with matching `applicationId` and `eventCode` and then extending the `time_scheduled` by the specified amount. It also simulates a trigger `ON_EXTEND` which can be used to trigger other events in the application (such as resetting the outcome from "EXPIRED" back to "PENING" for example).
+
+##### REQUEST parameters:
+
+- `applicationId`
+- `eventCode` -- value in `trigger_schedule` to match. For applicant deadlines we are currently using the hard-coded value `applicantDeadline` to match in the front-end UI. This event code is also passed along with the "ON_EXTEND" trigger to use to match to actions. 
+- `extensionTime` -- a length of time: either a number (which will be interpreted as days), or a [Luxon duration object](https://moment.github.io/luxon/api-docs/index.html#duration). If the current deadline has already expired, this extension time will be added to the current time. If it's not already expired, it'll be added to the current `time_scheduled`.
```
Comment (nmadruga) [inline documentation/API.md:456]:
```suggestion
- `extensionTime` -- a length of time: either a number (which will be interpreted as days), or a [Luxon duration object](https://moment.github.io/luxon/api-docs/index.html#duration). If the current deadline has already expired, this extension time will be added to the current time. If it's not already expired, it'll be added to the `time_scheduled`.
```

File: documentation/API.md [hunk 3]
```diff
@@ -352,6 +383,88 @@ For displaying custom data (e.g. Users, Products, Orgs). User's JWT determines w
...
+
+
+
+### Extend application deadline
+
+POST: `/extend-application`
+
+Endpoint for extending a deadline associated with an application. Currently, the main use case is for extending an applicant deadline for responding to a request for changes (event code: `applicantDeadline`), but in theory could be used for extending other types of deadlines as well.
+
+It works by finding an event in the `trigger_schedule` table with matching `applicationId` and `eventCode` and then extending the `time_scheduled` by the specified amount. It also simulates a trigger `ON_EXTEND` which can be used to trigger other events in the application (such as resetting the outcome from "EXPIRED" back to "PENING" for example).
```
Comment (nmadruga) [inline documentation/API.md:450]: I'm not entirely sure on this sentence. Maybe could use some update to be clearer.

Particularly this sentence: 
> It also simulates a trigger `ON_EXTEND` which can be used to trigger other events in the application (such as resetting the outcome from "EXPIRED" back to "PENING" for example).

**Review by nmadruga — CHANGES_REQUESTED** [review]
Just missing details for `Preview` action. I think for completeness it would be good to add the heading for each new action with a TODO so we know where to look

File: documentation/API.md (hunk 2 above)
Comment (CarlosNZ) [reply documentation/API.md]: Okay, I think this is clearer now: cf87a08e

File: documentation/API.md (hunk 3 above)
Comment (CarlosNZ) [reply documentation/API.md]: Cool -- how's this? 9a6ad9dd

Comment (nmadruga) [reply documentation/API.md]: I think it's easier to understand. But we should probably go through all the documentation at some point and refresh, specially when translating it to Portuguese. I think it's hard to understand a few concepts when it's very particular to one action.

Comment (CarlosNZ) [reply documentation/API.md]: > specially when translating it to Portuguese

This is developer reference docs -- would we ever translate this? (If I were writing it for general access, I'd write it completely differently, I'm mainly writing this stuff down so _I_ can remember it heh)

**Comment by CarlosNZ** [comment]
> Just missing details for `Preview` action. I think for completeness it would be good to add the heading for each new action with a TODO so we know where to look

All actions and triggers are documented. "Preview" is not an action, it's an endpoint which calls existing actions, and that's all documented. Let me know if there's anything missing.

**Review by nmadruga — APPROVED** [review]
All good

**Pull request merged**